src/scene/text-split/AbstractSplitText.ts

```diff
@@ -0,0 +1,554 @@
+     *     duration: 0.5,
+     *     delay: i * 0.2,
+     *     ease: 'back.out'
+     *   });
+     * });
+     * ```
+     */
+    public lines: Container[];
+
+    protected _originalText: string;
+    protected _lineAnchor: number | PointData;
+    protected _wordAnchor: number | PointData;
+    protected _charAnchor: number | PointData;
+    protected _autoSplit: boolean;
+    protected _style: TextStyle;
+
+    protected _dirty: boolean = false;
+    protected _canReuseChars: boolean = false;
+
+    constructor(config: AbstractSplitTextOptions)
+    {
+        const {
+            text,
+            style,
+            autoSplit,
+            lineAnchor,
+            wordAnchor,
+            charAnchor,
+            ...options
+        } = config;
+
+        super(options);
+        this.chars = [];
+        this.words = [];
+        this.lines = [];
+
+        this._originalText = text;
+        this._autoSplit = autoSplit;
+        this._lineAnchor = lineAnchor;
+        this._wordAnchor = wordAnchor;
+        this._charAnchor = charAnchor;
+
+        // setting the style will segment the text if autoSplit is true
+        this.style = style;
+    }
+
+    protected abstract splitFn(): TextSplitOutput<T>;
+
+    /**
+     * Splits the text into lines, words, and characters.
+     * Call this manually when autoSplit is false.
+     * @example Manual Splitting
+     * ```ts
+     * const text = new SplitText({
+     *   text: 'Manual Update',
+     *   autoSplit: false
+     * });
+     *
+     * text.text = 'New Content';
+     * text.style = { fontSize: 32 };
+     * text.split(); // Apply changes
+     * ```
+     */
+    public split(): void
+    {
+        const res: TextSplitOutput<T> = this.splitFn();
+
+        this.chars = res.chars;
+        this.words = res.words;
+        this.lines = res.lines;
+
+        this.addChild(...this.lines);
+
+        // force origin to be set
+        this.charAnchor = this._charAnchor;
+        this.wordAnchor = this._wordAnchor;
+        this.lineAnchor = this._lineAnchor;
+
+        this._dirty = false;
+        this._canReuseChars = true;
+    }
+
+    get text(): string
+    {
+        return this._originalText;
+    }
+    /**
+     * Gets or sets the text content.
+     * Setting new text triggers splitting if autoSplit is true.
+     * > [!NOTE] Setting this frequently can have a performance impact, especially with large texts and canvas text.
+     * @example Dynamic Text Updates
+     * ```ts
+     * const text = new SplitText({
+     *   text: 'Original',
+     *   autoSplit: true
+     * });
+     *
+     * // Auto-splits on change
+     * text.text = 'Updated Content';
+     *
+     * // Manual update
+     * text.autoSplit = false;
+     * text.text = 'Manual Update';
+     * text.split();
+     * ```
+     */
+    set text(value: string)
+    {
+        this._originalText = value;
+        this.lines.forEach((line) => line.destroy({ children: true }));
+        this.lines = [];
```

Comment:
this.lines.length = 0
```suggestion
        this.lines = [];
```